pkg/humio/client.go

```diff
@@ -237,27 +242,41 @@ func (c *Client) GetStream(method string, path string, query Query, ch chan Stre
 		return err
 	}
 	req = c.addAuthHeaders(req)
+
+	// Set up a context with timeout
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel() // Ensure cancellation to prevent resource leaks
+
+	req = req.WithContext(ctx)
+
 	res, err := c.StreamingClient.Do(req)
 	if err != nil {
 		return err
 	}
 	defer func() {
 		res.Body.Close()
 	}()
+
 	d := json.NewDecoder(res.Body)
 
-	// note from andrew: it might be a good idea to set up a ticker to not endlessly loop over this code
-	// you should look to see if this is done with the decoder already
-	// currently this never ends unless there is an error. there should be an ending condition for this loop
-	// maybe a timeout similar to the client timeout. or maybe we can check to see if the client is timed out. not sure
+	// Set up ticker to prevent infinite loop
```

Comment:
Its still an infinite loop. When i experimented with adding a ticker, it felt slow to get data back because it would get like 20 logs and then wait a second and get 20 more logs but that's probably fine as long as its returning all current (live) logs